src/Microsoft.Net.Http.Headers/HttpHeaderParser.cs

```diff
@@ -71,10 +81,15 @@ public virtual bool TryParseValues(IList<string> values, out IList<T> parsedValu
                             results.Add(output);
                         }
                     }
-                    else
+                    else if (strict)
                     {
                         return false;
                     }
+                    else
+                    {
+                        // Skip the invalid values and keep trying.
+                        index++;
```

Comment:
TryParse will update index to the first invalid character, so we skip it and try again.